Lesson 3 - exercises and assigment 1

anbaopham opened the pull request:
Lesson 3 - exercises and assigment 1

charlesrobison commented on students/anbaopham/Lesson 3/Assignment_1.py:
```diff
@@ -0,0 +1,124 @@
+            ("Mark Zuckerberg", [1663.23, 4300.87, 10432.0]),]
+
+def get_name_list():
+    name_list =[]
```
Missing space after assignment operator

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/Assignment_1.py:
```diff
@@ -0,0 +1,124 @@
+          "3 - Exit",
+          ">>> "))
+
+def add_donation_amount(FullName, donation_amount):
```
FullName should be snake_case: full_name

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/Assignment_1.py:
```diff
@@ -0,0 +1,124 @@
+def create_a_Report():
+    l = (["Donor Name", "Total Given", "Num Gifts", "Average Gift"])
+    temp_list = []
+    l1 =[]
```
Missing space here too

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/Assignment_1.py:
```diff
@@ -0,0 +1,124 @@
+    temp_list = []
+    l1 =[]
+
+    totalGiven =0
```
Need space after assignment operator

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/Assignment_1.py:
```diff
@@ -0,0 +1,124 @@
+        temp_list.append(aveGift)
+        #report_table.append(x[0], totalGiven, numGift, aveGift)
+    m = 0
+    while m< len(temp_list):
```
Missing space after 'm'

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/Assignment_1.py:
```diff
@@ -0,0 +1,124 @@
+
+    l1.insert(0,l)
+
+    newL = []
```
This should be snake case - like new_list or the like

charlesrobison commented on students/anbaopham/Lesson 3/Assignment_1.py:
```diff
@@ -0,0 +1,124 @@
+        l1.append(temp_list[m:m+4])
+        m = m+4
+
+    l1 =sorted(l1, key = sort_key)
```
Missing space

charlesrobison commented on students/anbaopham/Lesson 3/Assignment_1.py:
```diff
@@ -0,0 +1,124 @@
+
+if __name__ == "__main__":
+    # don't forget this block to guard against your code running automatically if this module is imported
+    main()
```
Missing a couple of elements:
1). The report should be sorted by total given in descending order
2). When adding a new donor and then running the report, the program errors and exits.  Please take another look as future versions build on this.

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_serries_3.py:
```diff
@@ -0,0 +1,40 @@
+
+fruit_list = ["Apples", "Pears","Oranges","Peaches"]
+
+updated_list =[]
```
Missing space

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_serries_3.py:
```diff
@@ -0,0 +1,40 @@
+                    updated_list.append(j)
+                else:
+                    pass
+            n = n+1
```
Missing spaces here too

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_serries_3.py:
```diff
@@ -0,0 +1,40 @@
+m = 0
+while m < 1:
+    for i in fruit_list:
+        askQ = input("do you love {} ? ".format(i).lower() )
```
askQ should be snake_case: ask_question

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_serries_3.py:
```diff
@@ -0,0 +1,40 @@
+            break
+
+if m > 0:
+    print(updated_list)
```
This loop never ends - did you run it successfully?

anbaopham replied:
When user type "x", m will be 1 so While loop will be end because m is not less than 1 any more.

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_serries_4.py:
```diff
@@ -0,0 +1,10 @@
+
+fruit_list = ["Apples", "Pears","Oranges","Peaches"]
```
Missing spaces after commas

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_serries_4.py:
```diff
@@ -0,0 +1,10 @@
+
+fruit_list = ["Apples", "Pears","Oranges","Peaches"]
+l = fruit_list[0:-1]
+newList = []
```
Should be snake_case

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_string_format_task1.py:
```diff
@@ -0,0 +1,19 @@
+def task_1(seq):
+    l = len(seq)-1
+    x=[]
```
Missing spaces

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_string_format_task1.py:
```diff
@@ -0,0 +1,19 @@
+    for i in seq[1::]:
+        n = format(i,'e')
+        x.append(n)
+    fileN =[]
```
Should be snake_case

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_string_format_task2.py:
```diff
@@ -0,0 +1,23 @@
+
+def task_2(seq):
+    l = len(seq)-1
+    x=[]
```
Missing spaces

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_string_format_task2.py:
```diff
@@ -0,0 +1,23 @@
+    for i in seq[1::]:
+        n = format(i,'e')
+        x.append(n)
+    fileN =[]
```
snake_case

charlesrobison replied:
And missing space

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_string_format_task2.py:
```diff
@@ -0,0 +1,23 @@
+        x.append(n)
+    fileN =[]
+    tempL = str(seq[0])
+    if len(tempL)==1:
```
Missing spaces

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_string_format_task2.py:
```diff
@@ -0,0 +1,23 @@
+        n = format(i,'e')
+        x.append(n)
+    fileN =[]
+    tempL = str(seq[0])
```
snake_case

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_string_format_task3.py:
```diff
@@ -0,0 +1,11 @@
+
+
+
+in_tuple = (1,3,5,7,8)
```
Missing spaces after commas

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_string_format_task4.py:
```diff
@@ -0,0 +1,11 @@
+    for i in seq:
+        #n = "{:02d}".format(i)
+        nums.append("{:02d}".format(i))
+    newNums = nums[-2::] + nums[2:-2] + nums[0:2]
```
snake_case

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_string_format_task4.py:
```diff
@@ -0,0 +1,11 @@
+    newNums = nums[-2::] + nums[2:-2] + nums[0:2]
+    print(newNums)
+
+seq=(4,30, 2017, 2,27)
```
Missing spaces

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_task_1.py:
```diff
@@ -0,0 +1,68 @@
+def exchange_first_last(seq):
+
+    a_new_squence = list(seq)
+    tempS = a_new_squence[0]
```
snake_case

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_task_1.py:
```diff
@@ -0,0 +1,68 @@
+
+    return remove_one(tempList)
+
+def sliceReversed(seq):
```
snake_case

charlesrobison commented on students/anbaopham/Lesson 3/lesson_3_task_1.py:
```diff
@@ -0,0 +1,68 @@
+        tempList ="".join(tempList)
+    return tempList
+
+def reOrder(seq):
```
snake_case

charlesrobison commented on students/anbaopham/Lesson_4/Assignment_2.py:
```diff
@@ -0,0 +1,177 @@
+
+if __name__ == "__main__":
+    # don't forget this block to guard against your code running automatically if this module is imported
+    main()
```
Same issues as prior version of mailroom.  Please clean up spacing and make sure variables and function names are snake_case.  Report needs to be sorted by total given in descending order.  Also, getting an error when adding a new donor to this version.

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson_4/Lesson_4_Trigrams.py:
```diff
@@ -0,0 +1,55 @@
+
+def build_trigrams(words):
+    trigrams = {}
+    value =[]
```
value assigned but not used

anbaopham replied:
deleted value = []

charlesrobison commented on students/anbaopham/Lesson_4/Lesson_4_Trigrams.py:
```diff
@@ -0,0 +1,55 @@
+    trigrams = {}
+    value =[]
+    random_choice = random.choice(words)
+    n =words.index(random_choice)
```
Missing space

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson_4/Lesson_4_Trigrams.py:
```diff
@@ -0,0 +1,55 @@
+
+
+def display_dictionary(trigrams):
+    for x,y in trigrams.items():
```
Missing space

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson_4/Lesson_4_Trigrams.py:
```diff
@@ -0,0 +1,55 @@
+
+def display_dictionary(trigrams):
+    for x,y in trigrams.items():
+        print(x,y)
```
Missing space

anbaopham replied:
fixed

charlesrobison commented on students/anbaopham/Lesson_4/Lesson_4_Trigrams.py:
```diff
@@ -0,0 +1,55 @@
+
+
+
+build_trigrams(words)
```
An - this isn't producing the expected results.  Is this code complete?

anbaopham replied:
I think it worked because it worked with "words" in line #3.  If we don't read "sherlock_small.txt", this code worked with "words" in line 3.
Did I miss something?

charlesrobison replied:
The output should be a created 'story' with sentences created by the functions in your code.

anbaopham replied:
If I run line #3, it showed as expected.  Can you give please more provide more detail how to improve my code?

charlesrobison replied:
The user should be able to pass in a file, like sherlock_small.txt, and the output should be a new, randomly generated short story.  Does that make sense.  I'm going to merge this PR, so please submit any new or edited code to a new PR for further review.

charlesrobison reviewed (approved):
Lots of PEP8 issues - please add a linter to your text editor.  Also, some elements missing or errors with mailroom assignments.  trigrams doesn't appear complete.  Please submit all of your lesson 3 assignments on the course site for graded.  Please reach out if you have questions.

Ready to merge

anbaopham commented:
Charles, 

I fixed all your comments about PEP8.

I uploaded lesson 3 in course website. 

What’s happening to lesson 4?  I don’t see it in Progress. 

Could you find all files for lesson 2?

Is there anything I can do to improve my grades in Progress?

Please let me know I missed anything. 

Thanks,

An. 



> On Sep 19, 2020, at 9:39 PM, Charles Robison <notifications@github.com> wrote:
> 
> ﻿
> @charlesrobison approved this pull request.
> 
> Lots of PEP8 issues - please add a linter to your text editor. Also, some elements missing or errors with mailroom assignments. trigrams doesn't appear complete. Please submit all of your lesson 3 assignments on the course site for graded. Please reach out if you have questions.
> 
> Ready to merge
> 
> In students/anbaopham/Lesson 3/Assignment_1.py:
> 
> > @@ -0,0 +1,124 @@
> +import sys
> +donor_db = [("William Gates, III", [653772.32, 12.17]),
> +            ("Jeff Bezos", [877.33]),
> +            ("Paul Allen", [663.23, 43.87, 1.32]),
> +            ("Mark Zuckerberg", [1663.23, 4300.87, 10432.0]),]
> +
> +def get_name_list():
> +    name_list =[]
> Missing space after assignment operator
> 
> In students/anbaopham/Lesson 3/Assignment_1.py:
> 
> > +        name_list.append(i[0])
> +    return name_list
> +
> +
> +def sort_key(donor_db):
> +    return donor_db[0].split(" ")[1]
> +
> +
> +prompt = "\n".join(("Welcome to the mailroom!",
> +          "Please choose from below options:",
> +          "1 - Send a Thank You",
> +          "2 - Create a Report",
> +          "3 - Exit",
> +          ">>> "))
> +
> +def add_donation_amount(FullName, donation_amount):
> FullName should be snake_case: full_name
> 
> In students/anbaopham/Lesson 3/Assignment_1.py:
> 
> > +        new_donor = []
> +        #donation_amount=int(input("Enter donation amount >> "))
> +        new_donor = (FullName,[donation_amount],)
> +        donor_db.append(new_donor)
> +
> +    else:
> +        add_donation_amount(FullName, donation_amount)
> +
> +
> +    print(f"Thank Mr/Ms {FullName.upper()} for donation $ {donation_amount}")
> +    print(donor_db)
> +
> +def create_a_Report():
> +    l = (["Donor Name", "Total Given", "Num Gifts", "Average Gift"])
> +    temp_list = []
> +    l1 =[]
> Missing space here too
> 
> In students/anbaopham/Lesson 3/Assignment_1.py:
> 
> > +        new_donor = (FullName,[donation_amount],)
> +        donor_db.append(new_donor)
> +
> +    else:
> +        add_donation_amount(FullName, donation_amount)
> +
> +
> +    print(f"Thank Mr/Ms {FullName.upper()} for donation $ {donation_amount}")
> +    print(donor_db)
> +
> +def create_a_Report():
> +    l = (["Donor Name", "Total Given", "Num Gifts", "Average Gift"])
> +    temp_list = []
> +    l1 =[]
> +
> +    totalGiven =0
> Need space after assignment operator
> 
> In students/anbaopham/Lesson 3/Assignment_1.py:
> 
> > +    totalGiven =0
> +    for x in donor_db:
> +        temp_list.append(x[0])
> +        tg = sum(list((x[1])))
> +        totalGiven = str(tg)
> +        temp_list.append(totalGiven)
> +        ng = len(x[1])
> +        numGift = str(ng)
> +        temp_list.append(numGift)
> +        #aveGift = str(round(totalGiven/numGift,2))
> +        aG = round(tg/ng,2)
> +        aveGift = str(aG)
> +        temp_list.append(aveGift)
> +        #report_table.append(x[0], totalGiven, numGift, aveGift)
> +    m = 0
> +    while m< len(temp_list):
> Missing space after 'm'
> 
> In students/anbaopham/Lesson 3/Assignment_1.py:
> 
> > +        temp_list.append(numGift)
> +        #aveGift = str(round(totalGiven/numGift,2))
> +        aG = round(tg/ng,2)
> +        aveGift = str(aG)
> +        temp_list.append(aveGift)
> +        #report_table.append(x[0], totalGiven, numGift, aveGift)
> +    m = 0
> +    while m< len(temp_list):
> +        l1.append(temp_list[m:m+4])
> +        m = m+4
> +
> +    l1 =sorted(l1, key = sort_key)
> +
> +    l1.insert(0,l)
> +
> +    newL = []
> This should be snake case - like new_list or the like
> 
> In students/anbaopham/Lesson 3/Assignment_1.py:
> 
> > +        totalGiven = str(tg)
> +        temp_list.append(totalGiven)
> +        ng = len(x[1])
> +        numGift = str(ng)
> +        temp_list.append(numGift)
> +        #aveGift = str(round(totalGiven/numGift,2))
> +        aG = round(tg/ng,2)
> +        aveGift = str(aG)
> +        temp_list.append(aveGift)
> +        #report_table.append(x[0], totalGiven, numGift, aveGift)
> +    m = 0
> +    while m< len(temp_list):
> +        l1.append(temp_list[m:m+4])
> +        m = m+4
> +
> +    l1 =sorted(l1, key = sort_key)
> Missing space
> 
> In students/anbaopham/Lesson 3/Assignment_1.py:
> 
> > +
> +        elif response == "2":
> +            create_a_Report()
> +        elif response == "3":
> +            exit_program()
> +        else:
> +            print("Not a valid option!")
> +
> +
> +
> +
> +
> +
> +if __name__ == "__main__":
> +    # don't forget this block to guard against your code running automatically if this module is imported
> +    main()
> Missing a couple of elements:
> 1). The report should be sorted by total given in descending order
> 2). When adding a new donor and then running the report, the program errors and exits. Please take another look as future versions build on this.
> 
> In students/anbaopham/Lesson 3/lesson_3_serries_3.py:
> 
> > @@ -0,0 +1,40 @@
> +
> +fruit_list = ["Apples", "Pears","Oranges","Peaches"]
> +
> +updated_list =[]
> Missing space
> 
> In students/anbaopham/Lesson 3/lesson_3_serries_3.py:
> 
> > @@ -0,0 +1,40 @@
> +
> +fruit_list = ["Apples", "Pears","Oranges","Peaches"]
> +
> +updated_list =[]
> +def remove_fruit(deleted_fruit, newList, updated_list):
> +    n =0
> +    for i in newList:
> +        if i == deleted_fruit:
> +            for j in newList:
> +                if j != deleted_fruit:
> +                    updated_list.append(j)
> +                else:
> +                    pass
> +            n = n+1
> Missing spaces here too
> 
> In students/anbaopham/Lesson 3/lesson_3_serries_3.py:
> 
> > +def remove_fruit(deleted_fruit, newList, updated_list):
> +    n =0
> +    for i in newList:
> +        if i == deleted_fruit:
> +            for j in newList:
> +                if j != deleted_fruit:
> +                    updated_list.append(j)
> +                else:
> +                    pass
> +            n = n+1
> +            break
> +    return updated_list, n
> +m = 0
> +while m < 1:
> +    for i in fruit_list:
> +        askQ = input("do you love {} ? ".format(i).lower() )
> askQ should be snake_case: ask_question
> 
> In students/anbaopham/Lesson 3/lesson_3_serries_3.py:
> 
> > +
> +        elif askQ.lower() == "yes" or askQ.lower() == "y":
> +            print("good")
> +            pass
> +
> +        elif askQ.lower() == "no" or askQ.lower() == "n":
> +            remove_fruit(i,fruit_list,updated_list)
> +        elif askQ.lower() =="exit" or askQ.lower() == "x":
> +            m = m+1
> +            break
> +        else:
> +            print("not good")
> +            break
> +
> +if m > 0:
> +    print(updated_list)
> This loop never ends - did you run it successfully?
> 
> In students/anbaopham/Lesson 3/lesson_3_serries_4.py:
> 
> > @@ -0,0 +1,10 @@
> +
> +fruit_list = ["Apples", "Pears","Oranges","Peaches"]
> Missing spaces after commas
> 
> In students/anbaopham/Lesson 3/lesson_3_serries_4.py:
> 
> > @@ -0,0 +1,10 @@
> +
> +fruit_list = ["Apples", "Pears","Oranges","Peaches"]
> +l = fruit_list[0:-1]
> +newList = []
> Should be snake_case
> 
> In students/anbaopham/Lesson 3/lesson_3_string_format_task1.py:
> 
> > @@ -0,0 +1,19 @@
> +def task_1(seq):
> +    l = len(seq)-1
> +    x=[]
> Missing spaces
> 
> In students/anbaopham/Lesson 3/lesson_3_string_format_task1.py:
> 
> > @@ -0,0 +1,19 @@
> +def task_1(seq):
> +    l = len(seq)-1
> +    x=[]
> +    for i in seq[1::]:
> +        n = format(i,'e')
> +        x.append(n)
> +    fileN =[]
> Should be snake_case
> 
> In students/anbaopham/Lesson 3/lesson_3_string_format_task2.py:
> 
> > @@ -0,0 +1,23 @@
> +
> +
> +
> +def task_2(seq):
> +    l = len(seq)-1
> +    x=[]
> Missing spaces
> 
> In students/anbaopham/Lesson 3/lesson_3_string_format_task2.py:
> 
> > @@ -0,0 +1,23 @@
> +
> +
> +
> +def task_2(seq):
> +    l = len(seq)-1
> +    x=[]
> +    for i in seq[1::]:
> +        n = format(i,'e')
> +        x.append(n)
> +    fileN =[]
> snake_case
> 
> In students/anbaopham/Lesson 3/lesson_3_string_format_task2.py:
> 
> > @@ -0,0 +1,23 @@
> +
> +
> +
> +def task_2(seq):
> +    l = len(seq)-1
> +    x=[]
> +    for i in seq[1::]:
> +        n = format(i,'e')
> +        x.append(n)
> +    fileN =[]
> +    tempL = str(seq[0])
> +    if len(tempL)==1:
> Missing spaces
> 
> In students/anbaopham/Lesson 3/lesson_3_string_format_task2.py:
> 
> > @@ -0,0 +1,23 @@
> +
> +
> +
> +def task_2(seq):
> +    l = len(seq)-1
> +    x=[]
> +    for i in seq[1::]:
> +        n = format(i,'e')
> +        x.append(n)
> +    fileN =[]
> +    tempL = str(seq[0])
> snake_case
> 
> In students/anbaopham/Lesson 3/lesson_3_string_format_task3.py:
> 
> > @@ -0,0 +1,11 @@
> +
> +def formatter(in_tuple):
> +    l = len(in_tuple)
> +    form_string = "{:d}"
> +    #print(("file_{}: " + ", ".join(["{}"]*l)).format(fileN, *x))
> +    print(f"the {l} numbers are: " + ", ".join(["{}"]*l).format(*in_tuple))
> +
> +
> +
> +in_tuple = (1,3,5,7,8)
> Missing spaces after commas
> 
> In students/anbaopham/Lesson 3/lesson_3_string_format_task4.py:
> 
> > @@ -0,0 +1,11 @@
> +
> +def task_4(seq):
> +    nums = []
> +    for i in seq:
> +        #n = "{:02d}".format(i)
> +        nums.append("{:02d}".format(i))
> +    newNums = nums[-2::] + nums[2:-2] + nums[0:2]
> snake_case
> 
> In students/anbaopham/Lesson 3/lesson_3_string_format_task4.py:
> 
> > @@ -0,0 +1,11 @@
> +
> +def task_4(seq):
> +    nums = []
> +    for i in seq:
> +        #n = "{:02d}".format(i)
> +        nums.append("{:02d}".format(i))
> +    newNums = nums[-2::] + nums[2:-2] + nums[0:2]
> +    print(newNums)
> +
> +seq=(4,30, 2017, 2,27)
> Missing spaces
> 
> In students/anbaopham/Lesson 3/lesson_3_task_1.py:
> 
> > @@ -0,0 +1,68 @@
> +def exchange_first_last(seq):
> +
> +    a_new_squence = list(seq)
> +    tempS = a_new_squence[0]
> snake_case
> 
> In students/anbaopham/Lesson 3/lesson_3_task_1.py:
> 
> > +    if isinstance(seq,str):
> +        newList ="".join(newList)
> +
> +    return newList
> +
> +def remove_one_4(seq):
> +    l = list(seq)
> +
> +    tempList = l[4:len(l)-4]
> +
> +    if isinstance(seq,str):
> +        tempList ="".join(tempList)
> +
> +    return remove_one(tempList)
> +
> +def sliceReversed(seq):
> snake_case
> 
> In students/anbaopham/Lesson 3/lesson_3_task_1.py:
> 
> > +
> +    tempList = l[4:len(l)-4]
> +
> +    if isinstance(seq,str):
> +        tempList ="".join(tempList)
> +
> +    return remove_one(tempList)
> +
> +def sliceReversed(seq):
> +    l = list(seq)
> +    tempList = l[::-1]
> +    if isinstance(seq,str):
> +        tempList ="".join(tempList)
> +    return tempList
> +
> +def reOrder(seq):
> snake_case
> 
> In students/anbaopham/Lesson_4/Assignment_2.py:
> 
> > +    while True:
> +        response = input(prompt)  # continuously collect user selection
> +        # now redirect to feature functions based on the user selection
> +        if response in response_dict:
> +            response_dict.get(response)()
> +        else:
> +            print("Not a valid option!")
> +
> +
> +
> +
> +
> +
> +if __name__ == "__main__":
> +    # don't forget this block to guard against your code running automatically if this module is imported
> +    main()
> Same issues as prior version of mailroom. Please clean up spacing and make sure variables and function names are snake_case. Report needs to be sorted by total given in descending order. Also, getting an error when adding a new donor to this version.
> 
> In students/anbaopham/Lesson_4/Lesson_4_Trigrams.py:
> 
> > +    if key in trigrams:
> +        return True
> +    else:
> +        return False
> +def get_value(words, pair):
> +    value_list = []
> +    for i in range(len(words)-2):
> +        if words[i] == pair[0] and words[i:i+2] == pair:
> +            value_list.append(words[i+len(pair)])
> +    return value_list
> +
> +
> +
> +def build_trigrams(words):
> +    trigrams = {}
> +    value =[]
> value assigned but not used
> 
> In students/anbaopham/Lesson_4/Lesson_4_Trigrams.py:
> 
> > +    else:
> +        return False
> +def get_value(words, pair):
> +    value_list = []
> +    for i in range(len(words)-2):
> +        if words[i] == pair[0] and words[i:i+2] == pair:
> +            value_list.append(words[i+len(pair)])
> +    return value_list
> +
> +
> +
> +def build_trigrams(words):
> +    trigrams = {}
> +    value =[]
> +    random_choice = random.choice(words)
> +    n =words.index(random_choice)
> Missing space
> 
> In students/anbaopham/Lesson_4/Lesson_4_Trigrams.py:
> 
> > +        key = " ".join(pair)
> +        follower = words[i+2]
> +
> +        if check_dictionary(trigrams, key) is False:
> +            trigrams[key] = follower
> +    for key in trigrams:
> +        pair = key.split()
> +        trigrams[key] = get_value(words, pair)
> +    print(random_choice)
> +    print(n)
> +    display_dictionary(trigrams)
> +    return trigrams
> +
> +
> +def display_dictionary(trigrams):
> +    for x,y in trigrams.items():
> Missing space
> 
> In students/anbaopham/Lesson_4/Lesson_4_Trigrams.py:
> 
> > +        follower = words[i+2]
> +
> +        if check_dictionary(trigrams, key) is False:
> +            trigrams[key] = follower
> +    for key in trigrams:
> +        pair = key.split()
> +        trigrams[key] = get_value(words, pair)
> +    print(random_choice)
> +    print(n)
> +    display_dictionary(trigrams)
> +    return trigrams
> +
> +
> +def display_dictionary(trigrams):
> +    for x,y in trigrams.items():
> +        print(x,y)
> Missing space
> 
> In students/anbaopham/Lesson_4/Lesson_4_Trigrams.py:
> 
> > +    for key in trigrams:
> +        pair = key.split()
> +        trigrams[key] = get_value(words, pair)
> +    print(random_choice)
> +    print(n)
> +    display_dictionary(trigrams)
> +    return trigrams
> +
> +
> +def display_dictionary(trigrams):
> +    for x,y in trigrams.items():
> +        print(x,y)
> +
> +
> +
> +build_trigrams(words)
> An - this isn't producing the expected results. Is this code complete?
> 
> In students/anbaopham/Lesson 3/lesson_3_string_format_task2.py:
> 
> > @@ -0,0 +1,23 @@
> +
> +
> +
> +def task_2(seq):
> +    l = len(seq)-1
> +    x=[]
> +    for i in seq[1::]:
> +        n = format(i,'e')
> +        x.append(n)
> +    fileN =[]
> And missing space
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

natasha-aleksandrova commented:
@charlesrobison could you review again plz?

charlesrobison commented on students/anbaopham/lesson05/Assignment_3.py:
```diff
@@ -0,0 +1,195 @@
+
+if __name__ == "__main__":
+    # don't forget this block to guard against your code running automatically if this module is imported
+    main()
```
Same issues with this version of mailroom as prior.  All functions should be lowercase with words separated by underscores. Report should be sorted by total given in descending order.  Also, when selecting report after adding donor, code is failing and stops running.

anbaopham replied:
I added sort_total_given function.  Somehow it doesn't sort as expected.  What should I do?

Did you mean that the code failed when you selected option 1 then select option 2?  I tested it and don't see error.  Please review it again or let me know what line cause the failure.  Thanks.

charlesrobison reviewed (approved):
Reviewed updates - please let me know if you have questions.

Ready to merge
